### PR TITLE
feat(placement): add top banner support

### PR DESCRIPTION
**Pull request by fiji-flo**
## Summary

Initial support for another placement.

---

### Before

image

### After

image

---

## How did you test this change?

Deploying to stage.

### CODE REVIEW

File: client/src/app.tsx [hunk 1]
```diff
@@ -34,25 +34,24 @@ const Sitemap = React.lazy(() => import("./sitemap"));
...
-  const isServer = useIsServer();
-
   React.useEffect(() => {
     setCategory(getCategoryByPathname(pathname));
   }, [pathname]);
 
   return (
     <>
       <A11yNav />
-      {!isServer && <Banner />}
```
Comment (fiji-flo) [inline client/src/app.tsx:50]: This is intentional and there will be a follow up PR.

File: libs/pong/get.js [hunk 2]
```diff
@@ -102,21 +101,36 @@ export function createPongGetHandler(client, coder, env) {
...
+              ];
+            } else if (p === "topBanner") {
+              const [{ contents, clickUrl, impressionUrl }] = v;
+              const { colors, cta } = contents?.[0]?.data?.customData || {};
+              return [
+                p,
+                {
+                  status: "success",
+                  copy: he.decode(contents?.[0]?.data?.title || ""),
+                  cta: he.decode(cta || "No CTA"),
```
Comment (caugner) [inline libs/pong/get.js:126]: This means all top placements will have a CTA, or else "No CTA" is shown?

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM overall (didn't test yet), but I have some comments.

File: client/src/placement-context.tsx [hunk 3]
```diff
@@ -19,25 +19,41 @@ export interface Fallback {
...
 export interface PlacementStatus {
-  status: Status.success;
+  status: Status;
   click: string;
   view: string;
   copy?: string;
   image?: string;
   fallback?: Fallback;
+  cta?: string;
+  colors?: {
```
[the remark this answers is not in the document]
Comment (fiji-flo) [reply client/src/placement-context.tsx:30]: it's really only colors in there

File: libs/pong/get.js (hunk 2 above)
Comment (fiji-flo) [reply libs/pong/get.js:126]: yes

File: client/src/placement-context.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (caugner) [reply client/src/placement-context.tsx:30]: Fair enough. It's just that `colors.color` is weird.

File: client/src/ui/organisms/placement/index.scss [hunk 4]
```diff
@@ -145,6 +145,7 @@ section.place {
   z-index: var(--z-index-top);
 
   &.empty {
+    border: none;
```
Comment (caugner) [inline client/src/ui/organisms/placement/index.scss:148]: Maybe rather move `border: 1px solid var(--border-primary);` inside `&.visible`?

**Review by caugner — APPROVED** [review]
Looking great 🚀

Edit: I'm a bit worried about the layout shift, but the only solution would be either SSR, or reserving the space by default depending on whether we received placement data previously.

**Pull request merged**